api/src/utils/validate-query.ts

```diff
@@ -101,6 +100,61 @@ function validateFilter(filter: Query['filter']) {
 	}
 }
 
+function validateNestedFilter(filter: Query['filter']) {
```

Comment:
I think that `validateFilter` should be refactored to accept a boolean such as `isNested`, as `validateNestedFilter` essentially the same with the removal of `_none` and `_some`.